riak_test/src/cs_suites.erl

```diff
@@ -0,0 +1,581 @@
+
+merge_all_bitcask(#state{riak_nodes=Nodes, riak_vsn=Vsn} = _State) ->
+    wait_until_merge_worker_idle(Nodes),
+    [trigger_bitcask_merge(N, Vsn) || N <- Nodes],
+    [ok = rt:wait_until(N,
+                        fun(Node) ->
+                                check_data_files_are_small_enough(Node, Vsn)
+                        end) || N <- Nodes],
+    ok.
+
+wait_until_merge_worker_idle(Nodes) ->
+    [ok = rt:wait_until(
+            N,
+            fun(Node) ->
+                    Status = rpc:call(Node, bitcask_merge_worker, status, []),
+                    lager:debug("Wait util bitcask_merge_worker to finish on ~p:"
+                                " status=~p~n", [Node, Status]),
+                    Status =:= {0, undefined}
+            end) || N <- Nodes],
+    ok.
+
+trigger_bitcask_merge(Node, Vsn) ->
+    lager:debug("Trigger bitcask merger for node ~s (version: ~s)~n", [Node, Vsn]),
+    VnodeNames = bitcask_vnode_names(Node, Vsn),
+    [begin
+         VnodeDir = filename:join(bitcask_data_root(), VnodeName),
+         DataFiles = bitcask_data_files(Node, Vsn, VnodeName, rel),
+         ok = rpc:call(Node, bitcask_merge_worker, merge,
+                       [VnodeDir, [], {DataFiles, []}])
+     end ||
+        VnodeName <- VnodeNames],
+    ok.
+
+bitcask_data_root() -> "./data/bitcask".
+
+%% Returns vnode dir name,
+%% For example: "548063113999088594326381812268606132370974703616"
+-spec bitcask_vnode_names(atom(), previous | current) -> [string()].
+bitcask_vnode_names(Node, Vsn) ->
+    Prefix = rtcs:get_rt_config(riak, Vsn),
+    BitcaskAbsRoot = rtcs:riak_bitcaskroot(Prefix, rt_cs_dev:node_id(Node)),
+    {ok, VnodeDirNames} = file:list_dir(BitcaskAbsRoot),
+    VnodeDirNames.
+
+%% Data file names should start bitcask's `data_root' for `bitcask_merge_worker'.
+%% This function returns a list proper for the aim, e.g.
+%% ["./data/bitcask/548063113999088594326381812268606132370974703616/5.bitcask.data",
+%%  "./data/bitcask/548063113999088594326381812268606132370974703616/6.bitcask.data"]
+-spec bitcask_data_files(node(), previous | current, string(), abs | rel) ->
+                                [string()].
+bitcask_data_files(Node, Vsn, VnodeName, AbsOrRel) ->
+    Prefix = rtcs:get_rt_config(riak, Vsn),
+    BitcaskAbsRoot = rtcs:riak_bitcaskroot(Prefix, rt_cs_dev:node_id(Node)),
+    VnodeAbsPath = filename:join(BitcaskAbsRoot, VnodeName),
+    {ok, Fs0} = file:list_dir(VnodeAbsPath),
+    [case AbsOrRel of
+         rel -> filename:join([bitcask_data_root(), VnodeName, F]);
+         abs -> filename:join([BitcaskAbsRoot, VnodeName, F])
+     end ||
+        F <- Fs0, filename:extension(F) =:= ".data"].
+
+%% Assert bitcask data is "small"
+%% 1) The number of *.data files should be =< 2
+%% 2) Each *.data file size should be < 32 KiB
+check_data_files_are_small_enough(Node, Vsn) ->
+    VnodeNames = bitcask_vnode_names(Node, Vsn),
+    check_data_files_are_small_enough(Node, Vsn, VnodeNames).
```

Comment:
`lists:all(fun(VNodeName) -> check_data_files_are_small_enough(Node, Vsn, VnodeName) end, VNodeNames).`

Hmm.. not so clearer for me :see_no_evil: The refactoring is left as an exercise for interested persons :sunglasses: